python/pip_install/tools/dependency_resolver/dependency_resolver.py

```diff
@@ -123,28 +127,28 @@ def _select_golden_requirements_file(
 
     sys.argv.append("--generate-hashes")
     sys.argv.append("--output-file")
-    sys.argv.append(requirements_txt if UPDATE else requirements_out)
+    sys.argv.append(requirements_file if UPDATE else requirements_out)
     sys.argv.append(
         requirements_in if requirements_in_path.exists() else resolved_requirements_in
     )
 
     if UPDATE:
```

Comment:
Thinking about the broader picture, we could have the `UPDATE` being handled by a separate script and use the approach elsewhere where:
1. There is a rule that generates the pinned `requirements_file`.
2. There is a `diff_test` rule, which checks the source version with the output of 1.
3. There is a rule that updates the source version.

Right now we use the same script for everything, which allows us to not use `bazel-skylib` for this functionality, but now we depend on it more heavily and thus we could refactor this code as well.

Whilst this suggestion is likely to be outside the scope of the PR, I would love to understand if the platform resolution complexity could be pushed to starlark and a bunch of `select` statements in the right places.

Reply:
It almost sound like you are aiming for how it is done in the pip_parse vendored example. To me it sounds like a sensible change.